sdks/java/core/src/main/java/org/apache/beam/sdk/io/FileBasedSink.java

```diff
@@ -341,8 +399,21 @@ public FileBasedSink(
 
   /** Return the {@link DynamicDestinations} used. */
   @SuppressWarnings("unchecked")
-  public <UserT> DynamicDestinations<UserT, DestinationT> getDynamicDestinations() {
-    return (DynamicDestinations<UserT, DestinationT>) dynamicDestinations;
+  public DynamicDestinations<UserT, DestinationT, OutputT> getDynamicDestinations() {
+    return (DynamicDestinations<UserT, DestinationT, OutputT>) dynamicDestinations;
+  }
+
+  /** Returns the list of side inputs used by this sink. */
+  public List<PCollectionView<?>> getSideInputs() {
+    return (sideInputs != null) ? sideInputs : dynamicDestinations.getSideInputs();
```

Comment:
This is here just for the translation code (so we can inject the side inputs deserialized from proto instead of calling to dynamicDestinations). Marking this as final, as we don't want users overriding this.